src/main/java/site/coach_coach/coach_coach_server/routine/controller/RoutineController.java

```diff
@@ -1,14 +1,59 @@
 package site.coach_coach.coach_coach_server.routine.controller;
 
+import java.util.List;
+
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+import lombok.RequiredArgsConstructor;
+import site.coach_coach.coach_coach_server.auth.userdetails.CustomUserDetails;
+import site.coach_coach.coach_coach_server.routine.dto.RoutineForListDto;
+import site.coach_coach.coach_coach_server.routine.dto.RoutineListCoachInfoDto;
+import site.coach_coach.coach_coach_server.routine.dto.RoutineListRequest;
+import site.coach_coach.coach_coach_server.routine.dto.RoutineListResponse;
+import site.coach_coach.coach_coach_server.routine.services.RoutineServices;
+
 @RestController
+@RequiredArgsConstructor
 @RequestMapping("/api")
 public class RoutineController {
+	private final RoutineServices routineServices;
+
+	@GetMapping("/v1/routines")
+	public ResponseEntity routines(@RequestParam(name = "coachId", required = false) Long coachId) {
+
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		Long userId = ((CustomUserDetails)authentication.getPrincipal()).getUserId();
+
+		RoutineListRequest routineListRequest = new RoutineListRequest(userId, coachId);
+
+		if (coachId == null) {
+			List<RoutineForListDto> routineListByMyself = routineServices.findRoutineForListServices(
+				routineListRequest);
+
+			return ResponseEntity.status(HttpStatus.OK).body(routineListByMyself);
+		} else {
+			Boolean checkMatching = routineServices.findIsMatchingServices(routineListRequest);
+			if (!checkMatching) {
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).body("아직 매칭이 되지 않은 코치입니다.");
+			} else {
+				RoutineListCoachInfoDto routineListCoachInfoDto = routineServices.findRoutineListCoachInfoServices(
+					routineListRequest);
+				List<RoutineForListDto> routineListByCoach = routineServices.findRoutineForListServices(
+					routineListRequest);
+
+				RoutineListResponse routineListResponse = new RoutineListResponse(routineListCoachInfoDto,
+					routineListByCoach);
+				return ResponseEntity.status(HttpStatus.OK).body(routineListResponse);
```

Comment:
이건 그냥 정보공유입니다 ㅎ
```suggestion
				return ResponseEntity.ok(routineListResponse);
```
이렇게도 가능합니다 ~~
멀고도 깊은 스프링 세계.... 됴르륵